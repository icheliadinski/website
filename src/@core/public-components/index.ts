export { About } from './about'
export { Contact } from './contact'
export { Blog } from './blog'
export { Skills } from './skills'
export { Works } from './works'
