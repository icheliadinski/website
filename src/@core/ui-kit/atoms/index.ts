export { Loader } from './loader'
export { ICON_TYPES, Icon } from './icon'
