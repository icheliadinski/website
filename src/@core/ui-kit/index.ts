export { Loader, Icon, ICON_TYPES } from './atoms'

export { LeftMenu } from './molecules'
