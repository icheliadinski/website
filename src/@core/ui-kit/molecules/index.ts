export { LeftMenu } from './left-menu'
