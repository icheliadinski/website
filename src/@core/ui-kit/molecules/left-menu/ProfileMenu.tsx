import React from 'react'
import classNames from 'classnames'
import { Icon, ICON_TYPES } from '@core/ui-kit'
import { noop } from '@core/libs'
import styles from './LeftMenu.module.scss'

export interface Props {
  icons?: ICON_TYPES[]
  activeItem?: ICON_TYPES
  onItemClick?: (itemType: ICON_TYPES | null) => void
}

export const ProfileMenu: React.FC<Props> = ({
  icons = [],
  activeItem,
  onItemClick = noop,
}) => {
  const handleItemClick = (itemType: ICON_TYPES) => onItemClick(itemType)

  const iconsList = icons.map(icon => {
    const classes = classNames(styles.item, {
      [styles.active]: activeItem === icon,
    })
    return (
      <li
        className={classes}
        key={icon}
        onClick={handleItemClick.bind(null, icon)}
        role="button"
      >
        <Icon type={icon} />
      </li>
    )
  })

  return <ul className={styles.profile}>{iconsList}</ul>
}
