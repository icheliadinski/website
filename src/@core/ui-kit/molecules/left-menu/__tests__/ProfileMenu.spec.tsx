import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import { ICON_TYPES } from '@core/ui-kit'
import { ProfileMenu, Props } from '../ProfileMenu'

describe('[ProfileMenu]', () => {
  let wrapper: ShallowWrapper<Props>

  beforeEach(() => {
    wrapper = shallow(<ProfileMenu />)
  })

  it('renders successfully', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('when has navigation items', () => {
    beforeEach(() => {
      wrapper.setProps({
        navigationItems: [
          { path: '', icon: ICON_TYPES.USER_ICON },
          { path: 'phone', icon: ICON_TYPES.PHONE_ICON },
        ],
      })
    })

    it('renders successfully', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
