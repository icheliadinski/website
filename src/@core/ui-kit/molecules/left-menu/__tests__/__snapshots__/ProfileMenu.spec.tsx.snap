// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ProfileMenu] renders successfully 1`] = `
<ul
  className="profile"
/>
`;

exports[`[ProfileMenu] when has navigation items renders successfully 1`] = `
<ul
  className="profile"
>
  <NavLink
    activeClassName="active"
    className="item"
    exact={true}
    key="user"
    role="button"
    to="/home/"
  >
    <Icon
      type="user"
    />
  </NavLink>
  <NavLink
    activeClassName="active"
    className="item"
    exact={true}
    key="phone"
    role="button"
    to="/home/phone"
  >
    <Icon
      type="phone"
    />
  </NavLink>
</ul>
`;
